templates/travel/partials/featured/featured.snip.html

```diff
@@ -0,0 +1,153 @@
+{{!
+<!---
+Copyright 2017 The AMP Start Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+}}
+
+<!-- Featured -->
+{{#partials.featured}}
+<section class="travel-featured pt3 relative clearfix">
+  <header class="max-width-2 mx-auto px1 md-px2 relative">
+    <h3 class="travel-featured-heading h1 bold line-height-2 mb2 center">Featured Destinations</h3>
+  </header>
+  <div class="max-width-3 mx-auto relative">
```

Comment:
there is a lot of repeating patterns in this file , all the data should go in the JSON and do a loop here so that we can see the markup clearly

Reply:
Fair- but we want to semantically declare how the images + content is imported into the markup, since there is a specific and varying amount of images that should be added to each column.

Reply:
Not sure what this exactly means. From what i see here if you have the right JSON it is possible to do a loop. Varying images can be an array?

Reply:
The grid is setup like so: 
```
<div>
        {{#image-col-1}}
        {{/image-col-1}}
        <div>
          {{#image-col-2-top}}
          {{/image-col-2-top}}
          {{#image-col-2-bottom}}
          {{/image-col-2-bottom}}
        </div>
      </div>
      <div>
        <div>
          {{#image-col-3-top}}
          {{/image-col-3-top}}
          {{#image-col-3-bottom}}
          {{/image-col-3-bottom}}
        </div>
        {{#image-col-4}}
        {{/image-col-4}}
      </div>
```

There is a div that wraps around the right two columns, a div that wraps around the left two columns, and a div that wraps around each column that has two images.

Reply:
```json
{
  "top-array": [
    {
     "top-outer-images": [],
     "inner-images": [],
     "bottom-outer-images": []
    }
  ]
}
```

I have given some really bad names above but its a concept that would work?

Reply:
Does `top-array` apply to the whole grid?

Reply:
its just a container object (please dont name it top array when you implement :))

Reply:
I dont see how this will work... I could see four loops potentially, one for each column.

Reply:
Is there a way we can revisit this after all other prs are merged?

Reply:
so your markup would be something like 
```mustache
      {{#top-array}}
      <div>
        {{#top-outer-images}}
        {{/top-outer-images}}
        <div>
          {{#inner-images}}
          {{/inner-images}}
        </div>
        {{#bottom-outer-images}}
        {{/bottom-outer-images}}
      </div>
      {{/top-array}}
```

Reply:
There is not a div that wraps around the whole grid, there are two sibling divs that wrap around the right two and left two columns respectively.

Reply:
To achieve what you described would also mean refactoring the markup to structure how the flex items work. To keep a good pace today, can we forgo this request? At least until later?

Reply:
Yes .